src/layout/Checkboxes/index.tsx

```diff
@@ -52,7 +54,42 @@ export class Checkboxes extends CheckboxesDef {
     return useEmptyFieldValidationOnlySimpleBinding(node);
   }
 
+  renderLayoutValidators(props: NodeValidationProps<'Checkboxes'>): JSX.Element | null {
+    return <CheckboxesLayoutValidator {...props} />;
+  }
+
   validateDataModelBindings(ctx: LayoutValidationCtx<'Checkboxes'>): string[] {
-    return this.validateDataModelBindingsSimple(ctx);
+    const errors: string[] = [];
+    const dataModelBindings = ctx.item.dataModelBindings ?? {};
+
+    if (!dataModelBindings?.group) {
+      const [newErrors] = this.validateDataModelBindingsSimple(ctx);
+      errors.push(...(newErrors || []));
+    }
+
+    const [newErrors] = this.validateDataModelBindingsAny(ctx, 'group', ['array'], false);
+    errors.push(...(newErrors || []));
+
+    if (dataModelBindings?.group) {
+      const isCompatible = dataModelBindings?.simpleBinding?.field.includes(`${dataModelBindings.group.field}.`);
+
+      if (!isCompatible) {
+        errors.push(`simpleBinding must reference a field in group`);
+      }
+
+      const simpleBindingPath = dataModelBindings.simpleBinding?.field.split('.');
+      const groupBinding = ctx.lookupBinding(dataModelBindings?.group);
+      const items = groupBinding[0]?.items;
+      const properties =
+        items && !Array.isArray(items) && typeof items === 'object' && 'properties' in items
+          ? items.properties
+          : undefined;
+
+      if (!(properties && simpleBindingPath[1] in properties)) {
+        errors.push(`The property ${simpleBindingPath[1]} must be present in group`);
+      }
```

Comment:
This also only supports one level deep structures. I recently fixed the validatior in `src/layout/List/index.tsx` to work with any levels of properties, you could take a look at how that does it.